### PR TITLE
Setup Project for Deployment

### PR DESCRIPTION
**Pull request by maiahta**

### CODE REVIEW

File: github/workflows/node.js.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+name: Node.js CI/CD
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
```
Comment (gokcedilek) [inline github/workflows/node.js.yaml:6]: shouldnt we only deploy on push, and not pull request?

File: client/src/components/CreateForm/CreateForm.js [hunk 2]
```diff
@@ -126,7 +126,7 @@ const CreateForm = ({ formType, onSuccess, onError, onClose, tripId }) => {
   const currentUser = useSelector((state) => state.get('auth').user).username;
 
   useEffect(() => {
-    axios.get('http://localhost:3001/user').then(
+    axios.get('/user').then(
```
Comment (gokcedilek) [inline client/src/components/CreateForm/CreateForm.js:129]: where did you set the axios base URL? is there a conditional logic such as 
```
axios.defaults.baseURL =
  process.env.NODE_ENV === 'development'
    ? process.env.REACT_APP_BE_URL
    : 'https://xyz.herokuapp.com';
```

**Comment by gokcedilek** [comment]
LGTM, thank you Maia!

File: client/src/components/CreateForm/CreateForm.js (hunk 2 above)
Comment (TheAvi123) [reply client/src/components/CreateForm/CreateForm.js:129]: I didn't have to do this for my assignment. Axios seems to manage this by itself.

File: github/workflows/node.js.yaml (hunk 1 above)
Comment (TheAvi123) [reply github/workflows/node.js.yaml:6]: I think we need both, PR commits don't count as pushes I think. That would be directly pushing code from the branch (through CLI), which I have disabled for the repo.

File: bin/www [hunk 3]
```diff
@@ -12,7 +12,7 @@ var http = require('http');
  * Get port from environment and store in Express.
  */
 
-var port = normalizePort(process.env.PORT || '3001');
+var port = normalizePort(process.env.PORT || '3008');
```
Comment (TheAvi123) [inline bin/www:15]: Just wondering why this was changed to 3008? Very specific number and I didn't need to change mine.

**Review by TheAvi123 — APPROVED** [review]

File: bin/www (hunk 3 above)
Comment (maiahta) [reply bin/www:15]: ah no reason, I stop and restart the server many times to test and I have to change the port everytime 😅 I'll change it back

File: client/src/components/CreateForm/CreateForm.js (hunk 2 above)
Comment (maiahta) [reply client/src/components/CreateForm/CreateForm.js:129]: yep. axios manages this by itself

**Pull request merged**